internal/builders/docker/pkg/builder.go

```diff
@@ -476,34 +478,74 @@ func (c *GitClient) checkoutGitCommit() error {
 	return nil
 }
 
+type tempFileResult struct {
+	File *os.File
+	Err  error
+}
+
+// A helper function used by saveToTempFile to process one individual file.
+func saveOneTempFile(verbose bool, reader io.Reader, fileChannel chan tempFileResult, printChannel chan string) {
+	var allBytes []byte
```

Comment:
I don't know if it's good practice or not, but you could:
```golang
defer fileChannel.Close()
defer printChannel.Close()
```

Reply:
We shouldn't close the channels here, because they are passed to more than one goroutine running this function. They should be only closed after the WaitGroup representing all active coroutines moves to the "Done" state.